### PR TITLE
speed-up testing

### PR DESCRIPTION
**Pull request by Borda**
I have introduced a smaller test dataset which is only 2k samples from MNIST dataset which lower the testing time from about 20min just to 10min :smiling_imp: 
I have also refactored imports in the test section

### CODE REVIEW

**Comment by williamFalcon** [comment]
@Borda can we spend some time working on making the tests faster? maybe use a smaller dataset or something?

**Comment by Borda** [comment]
> can we spend some time working on making the tests faster? maybe use a smaller dataset or something?

that would be perfect, (that is why I made this as WIP) since some contributors are creating PR with their first commit instead of developing in own branch and make PR when it is ready... the problem is that such approach takes almost all your CI workers so than all other PRs have to queue for hours...

Yes, in general the test should be as simple and straight forward as possible, so for example we can do:
- make the test dataset smaller
- have lower number of iteration
- use higher stop threshold

**Comment by Borda** [comment]
Screenshot at 2019-11-13 13-38-40
the extended timeout helped as expected, but still, we have failing tests for 1.1 and 1.2

**Comment by jeffling** [comment]
For the queuing problem: An alternative solution I've seen for bigger opensource projects would be to have a trigger for testing through a github bot. This way we don't run on every commit. 

1. User creates WIP PR, asks for advice
2. Advice administered
3. User finishes PR, comments '/run-ci'
4. A bot triggers CI on the build
5. Build passes and PR can be merged.

**Comment by Borda** [comment]
@williamFalcon test time 20min -> 10min

**Comment by Borda** [comment]
@williamFalcon @jeffling @neggert @Ir1d  test time 20min -> 10min

File: pytorch_lightning/testing/model_base.py [hunk 1]
```diff
@@ -19,6 +19,22 @@
...
+    def __init__(self, root, train=True, transform=None, target_transform=None,
+                 download=False, num_samples=8000):
+        super(TestingMNIST, self).__init__(
+            root,
+            train=train,
+            transform=transform,
+            target_transform=target_transform,
+            download=download
+        )
+        # take jsut a subset of MNIS dataset
```
Comment (Ir1d) [inline pytorch_lightning/testing/model_base.py]:
```suggestion
        # take just a subset of MNIST dataset
```

File: tests/test_amp.py [hunk 2]
```diff
@@ -9,20 +9,23 @@
     LightningTestModel,
 )
 from pytorch_lightning.utilities.debugging import MisconfigurationException
-from . import testing_utils
+from .utils import (
```
Comment (williamFalcon) [inline tests/test_amp.py]: it's clear to not import individual functions like this. It should be clear where this function lives and it's in testing_utils, not in this file.

Let's go back to:
```
testing_utils.some_func()
```

**Comment by williamFalcon** [comment]
@Borda great job!

**Pull request merged**